src/Core/Migration/Migration1582724349294AddNetAndGrossPurchasePrice.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+class Migration1582724349294AddNetAndGrossPurchasePrice extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1582724349294;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $this->addPurchasePriceFieldToOrderLineItems($connection);
+        $this->migrateProductPurchasePriceField($connection);
+        $this->addSynchDatabaseTrigger($connection);
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+        // nth
+    }
+
+    private function addPurchasePriceFieldToOrderLineItems(Connection $connection): void
+    {
+        $connection->executeQuery('ALTER TABLE `order_line_item` ADD `purchase_price` JSON NULL AFTER `price`;');
+    }
+
+    private function migrateProductPurchasePriceField(Connection $connection): void
+    {
+        // Fetch all product and their current purchase prices
```

Comment:
Please test with 1 million Products and add a reasonable Limit and a for each to allow an update in chunks